File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int len = tokens.size();
        stack<int> st;

        for (int i = 0; i < len; i++) {
            string str = tokens[i];
            if (str == "+" || str == "-" || str == "*" || str == "/") {
                int b = (st.top()); st.pop();
                int a = (st.top()); st.pop();

                if (str == "+") {
                    int c = a + b;
                    st.push(c);
                } else if (str == "-") {
                    int c = a - b;
                    st.push(c);
                } else if (str == "*") {
                    int c = a * b;
                    st.push(c);
                } else {
                    int c = a / b;
                    st.push(c);
                }

            } else {
                st.push(stoi(str));
            }
        }
        
        return st.top();
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        int n = tokens.length;

        Stack<Integer> st = new Stack<>();
        for (int i = 0; i < tokens.length; i++) {
            String token = tokens[i];
            if (token.equals("+") || token.equals("-") || token.equals("*") || token.equals("/")) {
                int op2 = st.pop();
                int op1 = st.pop();

                int res = 0;
                
                if (token.equals("+")) {
                    res = op1 + op2;
                } 

                if (token.equals("-")) {
                    res = op1 - op2;
                }

                if (token.equals("*")) {
                    res = op1 * op2;
                }

                if (token.equals("/")) {
                    res = op1 / op2;
                }

                st.push(res);
            } else {
                int op = Integer.parseInt(token);
                st.push(op);
            }
        }

        return st.peek();
        
    }
}